aws/aws-parallelcluster
# Add test for byos integration

Test does:
* get launch templates data and start a compute node
* test event handler execution and environment
* check artifacts are downloaded
* check artifacts created by head event are shared to compute
* check substack outputs are fetched
* check users are created and imds is set based on config
* check instance type data is fetched
* check cluster configuration is fetched
* check computes are terminated on cluster deletion

Tested with config:
```
{%- import 'common.jinja2' as common -%}
---
test-suites:
  byos:
    test_byos.py::test_byos:
      dimensions:
        - regions: ["eu-west-1"]
          instances: {{ common.INSTANCES_DEFAULT_X86 }}
          oss: ["alinux2"]
          schedulers: ["byos"]
```

Signed-off-by: Luca Carrogu <carrogu@amazon.com>

**Please See** [Git Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions)

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
